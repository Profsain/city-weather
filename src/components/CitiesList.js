import React from 'react';
import { useNavigate } from "react-router-dom";
import { Grid } from '@mui/material';
import { CityCard } from './CityCard';
import { nigerianCity } from '../helper/nigerianCity';
import '../styles/CitiesList.css';

const CitiesList = () => {
  const navigate = useNavigate();

  const loadWeatherDetails = (cityName) => {
    navigate("/weather", { state: cityName })
  }

  return (
    <Grid
      className="Cities-container"
      container
      direction="row"
      gap={2}
      justifyContent="center"
    >
      {nigerianCity.map((city) => (
        <Grid
          item xs={12}
          sm={2}
          key={city.id}
          onClick={() => loadWeatherDetails(city.name)}
        >
          <CityCard
            cityName={city.name}
            cityPic={city.image}
          />
        </Grid>
      ))}
      {/* <Grid item xs={12} sm={2}>
        <CityCard />
      </Grid>
      <Grid item xs={12} sm={2}>
        <CityCard />
      </Grid>
      <Grid item xs={12} sm={2}>
        <CityCard />
      </Grid>
      <Grid item xs={12} sm={2}>
        <CityCard />
      </Grid>
      <Grid item xs={12} sm={2}>
        <CityCard />
      </Grid>
      <Grid item xs={12} sm={2}>
        <CityCard />
      </Grid>
      <Grid item xs={12} sm={2}>
        <CityCard />
      </Grid> */}
    </Grid>
  )
};

export default CitiesList;
