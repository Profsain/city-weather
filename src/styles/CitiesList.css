.Cities-container {
  height: 100%;
  padding: 38px;
}
