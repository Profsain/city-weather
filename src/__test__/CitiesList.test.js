import React from 'react';
import { Router } from 'react-router-dom';
import {createMemoryHistory} from 'history';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configurestore';
import CitiesList from '../components/CitiesList';

it('Renders CitiesList component correctly', () => {
  const history = createMemoryHistory();

  const tree = renderer.create(
    <Provider store={store}>
      <Router location={history.location} navigator={history}>
        <CitiesList />
      </Router>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
