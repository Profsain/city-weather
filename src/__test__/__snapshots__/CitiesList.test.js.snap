// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders CitiesList component correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container Cities-container css-16wvrx1-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-2 css-63b4os-MuiGrid-root"
    onClick={[Function]}
  >
    <div
      className="City-card"
    >
      <img
        alt="abuja city"
        src="citiesImages/abuja.jpg"
      />
      <h4>
        Abuja
      </h4>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-2 css-63b4os-MuiGrid-root"
    onClick={[Function]}
  >
    <div
      className="City-card"
    >
      <img
        alt="abuja city"
        src="citiesImages/abia.jpg"
      />
      <h4>
        Abia
      </h4>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-2 css-63b4os-MuiGrid-root"
    onClick={[Function]}
  >
    <div
      className="City-card"
    >
      <img
        alt="abuja city"
        src="citiesImages/Benin.jpg"
      />
      <h4>
        Benin
      </h4>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-2 css-63b4os-MuiGrid-root"
    onClick={[Function]}
  >
    <div
      className="City-card"
    >
      <img
        alt="abuja city"
        src="citiesImages/enugu.jpg"
      />
      <h4>
        Enugu
      </h4>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-2 css-63b4os-MuiGrid-root"
    onClick={[Function]}
  >
    <div
      className="City-card"
    >
      <img
        alt="abuja city"
        src="citiesImages/ibadan.jpg"
      />
      <h4>
        Ibadan
      </h4>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-2 css-63b4os-MuiGrid-root"
    onClick={[Function]}
  >
    <div
      className="City-card"
    >
      <img
        alt="abuja city"
        src="citiesImages/ilorin.jpg"
      />
      <h4>
        Ilorin
      </h4>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-2 css-63b4os-MuiGrid-root"
    onClick={[Function]}
  >
    <div
      className="City-card"
    >
      <img
        alt="abuja city"
        src="citiesImages/jos.jpg"
      />
      <h4>
        Jos
      </h4>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-2 css-63b4os-MuiGrid-root"
    onClick={[Function]}
  >
    <div
      className="City-card"
    >
      <img
        alt="abuja city"
        src="citiesImages/kaduna.jpg"
      />
      <h4>
        Kaduna
      </h4>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-2 css-63b4os-MuiGrid-root"
    onClick={[Function]}
  >
    <div
      className="City-card"
    >
      <img
        alt="abuja city"
        src="citiesImages/kano.jpg"
      />
      <h4>
        Kano
      </h4>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-2 css-63b4os-MuiGrid-root"
    onClick={[Function]}
  >
    <div
      className="City-card"
    >
      <img
        alt="abuja city"
        src="citiesImages/lagos.jpg"
      />
      <h4>
        Lagos
      </h4>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-2 css-63b4os-MuiGrid-root"
    onClick={[Function]}
  >
    <div
      className="City-card"
    >
      <img
        alt="abuja city"
        src="citiesImages/maiduguri.jpg"
      />
      <h4>
        Maiduguri
      </h4>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-2 css-63b4os-MuiGrid-root"
    onClick={[Function]}
  >
    <div
      className="City-card"
    >
      <img
        alt="abuja city"
        src="citiesImages/Ph.jpg"
      />
      <h4>
        Port
      </h4>
    </div>
  </div>
</div>
`;
